src/main/java/spleetwaise/transaction/logic/commands/EditCommand.java

```diff
@@ -91,9 +91,10 @@ private static Transaction createEditedTransaction(
         Amount amount = editTransactionDescriptor.getAmount().orElse(txnToEdit.getAmount());
         Description description = editTransactionDescriptor.getDescription().orElse(txnToEdit.getDescription());
         Date date = editTransactionDescriptor.getDate().orElse(txnToEdit.getDate());
+        Status status = editTransactionDescriptor.getStatus().orElse(txnToEdit.getStatus());
         Set<Category> categories = editTransactionDescriptor.getCategories().orElse(txnToEdit.getCategories());
 
-        return new Transaction(id, person, amount, description, date, categories);
+        return new Transaction(id, person, amount, description, date, categories).setStatus(status);
```

Comment:
What is the consideration behind this line, i.e. status not included as the constructor but through the setStatus method?

Reply:
This is due to the assumption that all newly created transactions are **Undone**, thus to create a **Done** transaction, you have to first create a `Transaction` then `setStatus`. In new discussion, we agree that you can create new Done transaction straight away so this should be change in new PR alongside with `addTxn` command with status field.